autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +121,79 @@ def get_dataset(self, segment='train'):
         sampler, _ = get_split_samplers(dataset, split_ratio=0.2)
         return SampledDataset(dataset, sampler)
 
-class TSVClassificationTask(AbstractGlueTask):
-    def __init__(self, *args, **kwargs): # passthrough arguments to TSVDataset
-        # (filename, field_separator=nlp.data.Splitter(','), num_discard_samples=1, field_indices=[2,1])
-        self.args = args
-        self.kwargs = kwargs
-        is_pair = False
-        class_labels = ['0', '1']
-        metric = CompositeEvalMetric()
-        metric.add(F1())
-        metric.add(Accuracy())
-        super(TSVClassificationTask, self).__init__(class_labels, metric, is_pair)
-        dataset = nlp.data.TSVDataset(*self.args, **self.kwargs)
-        # do the split
-        train_sampler, val_sampler = get_split_samplers(dataset, split_ratio=0.8)
-        self.trainset = SampledDataset(dataset, train_sampler)
-        self.valset = SampledDataset(dataset, val_sampler)
+class CustomTSVClassificationTask(AbstractGlueTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
+        Any valid string path is acceptable.
+    sep : str
+        Delimiter to use.
+    header : int, list of int, default 'infer'
+        Row number(s) to use as the column names, and the start of the
+        data.
+    names : array-like, optional
+        List of column names to use.
+    index_col : int, str, sequence of int / str, or False, default ``None``
+    usecols : list-like or callable, optional
+        Return a subset of the columns.
+    For our keyword argument, plesase see
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    """
+    def __init__(self, *args, **kwargs):
+        self._read(**kwargs)
+        self.is_pair = False
+        self.class_labels = list(set([sample[1] for sample in self.dataset]))
+        self.metric = Accuracy()
+        super(CustomTSVClassificationTask, self).__init__(self.class_labels, self.metric, self.is_pair)
+
+    def get_dataset(self, segment='train'):
+        """Get the corresponding dataset for the task.
+
+        Parameters
+        ----------
+        segment : str, default 'train'
+            Dataset segments.
+
+        Returns
+        -------
+        The dataset of target segment.
+        """
+        if segment == 'train':
+            return SampledDataset(self.dataset, self.train_sampler)
+        elif segment == 'dev':
+            return SampledDataset(self.dataset, self.dev_sampler)
+        else:
+            raise NotImplementedError
+
+    def _read(self, **kwargs):
+        path = kwargs.get('filepath', None)
+        kwargs['filepath_or_buffer'] = path
+        kwargs.pop("filepath")
+        dataset_df = pd.read_csv(**kwargs)
```

Comment:
Add data loading specific parameter that houses the params, don't use kwargs here.

Reply:
Again, this is design preference. I would prefer such design and provides more detailed documentation to specify the valid user argument. Besides, it is internal implementation, users don't interact directly with the class.